### ISSUE TITLE
[issue] When publishing, an error from one package breaks the call to create a tag for all the others

### ISSUE THREAD

**Issue by ruscon** [changesets/action#218]
for example
2 packages should be published:
- `package A`
- `package B`

If for some reason the `package A` fails to publish to npm, but the `package B` publishes successfully, then the git tag and github release for the `package B` will not be generated.

**Comment by Andarist** [comment]
That is true - but it's also a little bit complicated to fix. Should `package B` even be considered to be successfully published in this case? What if it, let's say, depends on `package A`?

**Comment by ruscon** [comment]
@Andarist It would be cool if the packages were published one by one, taking into account the dependencies among themselves, but in this case, this is not important. Consider that packages do not depend on each other in this case.

**Comment by Andarist** [comment]
Well, this case is important for designing how it should work. I agree that this is overall a problem worth solving but to do that we need to consider all the possible situations.

**Comment by ruscon** [comment]
@Andarist if I think that everything should start with the fact that need a function to calculate the dependency tree. This exists, for example, in ([nx affected](https://nx.dev/concepts/affected). `nx affected` compute a list of changed packages based on the difference between two commits.
Would the same solution work here as well?

**Comment by bluwy** [comment]
Fixed in #535
